packages/tools/webpack/extract-i18n-expressions-webpack-plugin.js

```diff
@@ -91,16 +100,15 @@ module.exports = class ExtractI18nExpressionsWebpackPlugin {
 				compilation.getPath( '[file]', { filename: chunkJSFile } )
 			);
 
-			// Create Webpack source object which represents a new file, and add it to Webpack assets array.
-			compilation.assets[ assetFilename ] = new RawSource(
-				[ ...( translationCallExpressions.get( mainFilePath ) || new Set() ) ]
-					.map( ( expr ) => `${ expr };` )
-					.join( '' )
-			);
+			const isComment = ( expr ) => /^(\/\/|\/\*).+/.test( expr );
 
-			// Let Webpack know that the file that was created is a part of the chunk we're currently processing.
-			chunk.files.add( assetFilename );
-		} );
+			await fs.promises.writeFile(
+				path.join( compilation.options.output.path, assetFilename ),
+				( translationCallExpressions.get( mainFilePath ) || [] )
+					.map( ( expr ) => `${ expr }${ isComment( expr ) ? '' : ';' }` )
+					.join( '\n' )
+			);
+		}
```

Comment:
IDK if it'll help, but maybe you can use map instead of a for loop, and then `Promise,all()` to make it faster?

Reply:
Done